src/core_plugins/input_control_vis/public/control/control.js

```diff
@@ -7,17 +7,74 @@ Choose a different field or index documents that contain values for this field.`
 }
 
 export class Control {
-  constructor(controlParams, filterManager) {
+  constructor(controlParams, filterManager, kbnApi, useTimeFilter) {
     this.id = controlParams.id;
     this.options = controlParams.options;
     this.type = controlParams.type;
     this.label = controlParams.label ? controlParams.label : controlParams.fieldName;
+    this.useTimeFilter = useTimeFilter;
     this.filterManager = filterManager;
-    this.enable = true;
+    this.kbnApi = kbnApi;
+
     // restore state from kibana filter context
     this.reset();
+    // disable until initialized
+    this.disable('Control has not been initialized');
+  }
+
+  async fetch() {
+    throw new Error('fetch method not defined');
+  }
+
+  /**
+   *
+   * @param ancestors {array of Controls}
+   */
+  setAncestors(ancestors) {
+    this.ancestors = ancestors;
+  }
+
+  hasAncestors() {
+    return this.ancestors && this.ancestors.length > 0;
   }
 
+  hasUnsetAncestor() {
+    return this.ancestors.reduce((accumulator, ancestor) => {
+      return accumulator || !ancestor.hasValue();
+    }, false);
+  }
+
+  getAncestorSignature() {
```

Comment:
Why are we using string concatenation to determine object equality as opposed to using `_.isEqual`?

Reply:
I needed some way to store a hash of the last query. Saving the values seemed like an easy way to create a hash.

Reply:
Couldn't we just skip the `.join` then and compare the array of ancestor values?